### ISSUE TITLE
_NoLogging methods in VM

### ISSUE THREAD

**Issue by jasper-d** [dotnet/runtime#89634]
Since #68717 there are a bunch of `Xyz` and `Xyz_NoLogging` method pairs in VM code.
In most cases `Xyz` simply calls the _NoLogging variant or has the same implementation, e.g.
https://github.com/dotnet/runtime/blob/f5ca995bb6c9ab79993c3bd63a2391b393c4fd14/src/coreclr/vm/methodtable.h#L879-L890

I think the only methods that differ in behaviour are `MethodTable::GetClass_NoLogging`/`MethodTable::GetClass`.

Could these _NoLogging variants be removed or are they kept for some reason?

**Comment by AaronRobinsonMSFT** [comment]
> Could these _NoLogging variants be removed or are they kept for some reason?

These are an historical artifact for scenarios involving collection of data that were reused for performance related tooling. The "_NoLogging" APIs were lighter-weight in the sense they didn't record accesses as opposed to the "logging" versions that did. All/most of that infrastructure was removed but we kept the "_NoLogging" in case we find a use case for this sort of mechanism again.

I personally would be fine removing them since we are going to add new APIs that probably would need a log/no-log pair but since we don't measure or have an active scenario they won't get written. Which means we are going to need to audit everything again anyways.

@davidwrighton and @jkotas Do either of you have any push back on removing these?
